doc/sphinx/language/core/variants.rst

```diff
@@ -6,6 +6,10 @@ Variants and the `match` construct
 Variants
 --------
 
+The :cmd:`Variant` command allows to define types by case on the inhabitants of the type. This includes enumerated types from programming languages, such as Booleans, characters or even the degenerate cases of the unit and empty types. Variant types more generally include enumerated types with arguments or even enumerated types with parametric arguments such as option types, sum types, etc. It also includes predicates or type families defined by cases such as the Boolean reflection or equality predicates. Observing the form of the inhabitant of a variant type is done by case analysis using the :cmd:`match` expression. Each case is described by what is called a :g:`constructor` of the type. For instance, Booleans have two constructors: :g:`true` and :g:`false`.
```

Comment:
```suggestion
The :cmd:`Variant` command allows defining types by case on the inhabitants of the type. This includes enumerated types from programming languages, such as Booleans, characters or even the degenerate cases of the unit and empty types. Variant types more generally include enumerated types with arguments or even enumerated types with parametric arguments such as option types, sum types, etc. It also includes predicates or type families defined by cases such as the Boolean reflection or equality predicates. Observing the form of the inhabitant of a variant type is done by case analysis using the :cmd:`match` expression. Each case is described by what is called a :g:`constructor` of the type. For instance, Booleans have two constructors: :g:`true` and :g:`false`.
```

Variant types are not recursive (your examples boolean and character are not)?
Maybe add "inhabitant" to the glossary if it's essential to use that word here.
"Constructor" should be in the glossary and defined more explicitly and less tersely.

Please split into multiple lines.